pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,23 +19,49 @@ package azure
 import (
 	"fmt"
 
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
+const (
+	defaultStorageAccountType  = string(storage.StandardLRS)
+	fileShareAccountNamePrefix = "f"
+)
+
 // CreateFileShare creates a file share, using a matching storage account
 func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location string, requestGB int) (string, string, error) {
 	var err error
 	accounts := []accountWithLocation{}
-	if len(storageAccount) > 0 {
-		accounts = append(accounts, accountWithLocation{Name: storageAccount})
-	} else {
-		// find a storage account
-		accounts, err = az.getStorageAccounts()
-		if err != nil {
-			// TODO: create a storage account and container
-			return "", "", err
+	if len(storageAccount) == 0 {
+		// find a storage account that matches storageType
+		accounts, err = az.getStorageAccounts(storageType)
+		if err != nil || len(accounts) == 0 {
```

Comment:
if err!=nil then should return error directly?

Reply:
I think the error in that place is inaccurate. If there is no storage account in resource group. Then Value would be an empty array instead of nil. That's to prevent null reference error during iterating 'result.Value' later.

If result.Value is really nil, then we could think about server returning invalid response, which is considered to be an error.

Sample empty storage account response from server: `'{"value":[]}'`